api/src/main/java/org/openmrs/parameter/VisitSearchCriteriaBuilder.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.parameter;
+
+import org.openmrs.Concept;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.VisitAttributeType;
+import org.openmrs.VisitType;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * A builder class for constructing instances of {@link VisitSearchCriteria}.
+ * This builder allows for the flexible creation of {@link VisitSearchCriteria}
```

Comment:
https://openmrs.atlassian.net/wiki/spaces/docs/pages/25477044/Java+Conventions#New-Public-Methods-and-Classes